sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.getCompletedApplicationDetails.e2e-spec.ts

```diff
@@ -65,6 +75,52 @@ describe("ApplicationStudentsController(e2e)-getCompletedApplicationDetails", ()
       FakeStudentUsersTypes.FakeStudentUserType1,
       dataSource,
     );
+    student.sinValidation = createFakeSINValidation(
+      {
+        student,
+      },
+      { initialValue: { isValidSIN: true } },
+    );
+    await db.student.save(student);
+  });
+
+  beforeEach(async () => {
+    const studentRestrictions = await db.studentRestriction.find({
+      select: { id: true, student: { id: true } },
+      relations: { student: true },
+      where: { student: { id: student.id } },
+    });
+    await db.studentRestriction.remove(studentRestrictions);
+
+    // Arrange
+    // Create an application with the first disbursement status as pending for eCert validation test cases.
+    application = await saveFakeApplicationDisbursements(
+      appDataSource,
+      { student },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+        offeringIntensity: OfferingIntensity.partTime,
+        firstDisbursementInitialValues: {
+          coeStatus: COEStatus.completed,
+          disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+        },
+      },
+    );
+    [firstDisbursement] = application.currentAssessment.disbursementSchedules;
+    // Create a MSFAA number record for the first disbursement.
+    const msfaaNumber = createFakeMSFAANumber(
+      {
+        student,
+        referenceApplication: application,
+      },
+      {
+        msfaaState: MSFAAStates.Signed,
+      },
+    );
+    await db.msfaaNumber.save(msfaaNumber);
+
+    firstDisbursement.msfaaNumber = msfaaNumber;
+    await db.disbursementSchedule.save(firstDisbursement);
```

Comment:
Please avoid splitting the "Arrange" section is this way and please do not create methods that update file-scoped variables in the way that `application` and `firstDisbursement` are being updated here.